src/fs.js

```diff
@@ -0,0 +1,101 @@
+import { access, rm, writeFile } from 'fs/promises';
+import { rename as fsRename } from 'fs/promises';
+
+import { makeAbsolutePath, parsePath } from './utils/changePath.js';
+import { createReadStream, createWriteStream } from 'fs';
+import * as path from 'path';
+
+export async function runCat(catPath) {
+  let absPath;
+  try {
+    const parsedPath = parsePath(catPath);
+    absPath = makeAbsolutePath(parsedPath.argPath);
+    await access(absPath);
+
+    read(absPath);
+  } catch {
+    console.error(`"${absPath}" doesn't exist or you have no access`);
+  }
+}
+
+export async function runAdd(fileName) {
+  if (!fileName) {
+    throw new Error('Specify filename');
+  }
+  const absPath = path.join(process.cwd(), fileName);
+  await writeFile(absPath, '', { flag: 'wx+' });
+}
+
+export async function runRename(options) {
+  try {
+    const args = parsePath(options);
+    const filePath = makeAbsolutePath(args.argPath);
+    const restArgs = parsePath(args.argRest);
+    const fileName = restArgs.argPath;
+    try {
+      await access(filePath);
+    } catch {
+      throw new Error('operation failed');
+    }
+    const fileFolder = path.dirname(filePath);
+    const newPath = path.join(fileFolder, fileName);
+
+    await fsRename(filePath, newPath);
+    console.log(`file name successeful changed to ${fileName}`);
+  } catch (err) {
+    console.error(err.message);
+  }
+}
+
+export async function runCopy(options) {
+  try {
+    const args = parsePath(options);
+    const fromPath = makeAbsolutePath(args.argPath);
+    const restArgs = parsePath(args.argRest);
+    const toPath = path.join(makeAbsolutePath(restArgs.argPath), path.basename(fromPath));
+    await access(fromPath);
+    await access(makeAbsolutePath(restArgs.argPath));
+    const readStream = createReadStream(fromPath);
+    const writeStream = createWriteStream(toPath);
+    readStream.pipe(writeStream);
+    console.log('file successeful copied');
+  } catch {
+    console.error('operation failed');
+  }
+}
```

Comment:
This function doesn't work properly. I tried absolute path, relative path, but it doesn't work